import.sh

```diff
@@ -1,4 +1,35 @@
 #!/usr/bin/env sh
+#
+# This script is meant to be used to update the version of rocksdb in this repo.
+#
+# To use it, update the version number in the URL to the tar.gz file below, then
+# run it. If you're lucky, everything will just work, but you may get an error
+# about a patch not applying cleanly. If that's the case, try to fix up the
+# patch file so that it will apply cleanly, then reset the rest of the
+# repository and try again.
+#
+# Once things have run cleanly, you should have a fairly large diff of the
+# actual rocksdb code, potentially along with some added or removed symlink
+# files in the base of the repository. If a lot of symlink files are removed,
+# something has gone wrong, and you should try regenerating them using the loop
+# at the bottom of this script.
+#
+# Once you've successfully run this script, you should also manually verify that
+# the recommended build flags for darwin and linux mostly match up with what's
+# specified in the cgo_flags.go file in the base of the repo.
+# To do so go through the below process on both a mac machine and a linux
+# machine (or a linux docker container):
+#   cd internal
+#   ./build_tools/build_detect_platform tmp.txt
+#   # Manually compare the PLATFORM_CCFLAGS and PLATFORM_CXXFLAGS to what's in
+#   # cgo_flags.go
+#
+# Ask @tamird if you run into issues along the way.
+#
+# If you want to restore the repo to what it would look like when first
```

Comment:
Done.

But for my own understanding, is that the case even if you're doing the build/test by updating the commit hash in cockroachdb/cockroach/GLOCKFILE to the hash of the local commit in this repo? i.e. does the local build of cockroachdb just use whatever the current view of this repo is or does it first lock to the particular commit as part of the build process?